design/als-design.md

```diff
@@ -0,0 +1,68 @@
+# AccessLog Service (ALS) Support
+
+Status: Draft
+
+## Abstract
+
+This document proposed a design for supporting Envoy's AccessLog Service capabilities in Contour.
+
+## Background
+
+Envoy has the ability to send rich access logs to access logging sinks.
+With AccessLog Service (ALS) enabled, Envoy can send access log messages using gRPC streams to a gRPC access log service. 
+
+This document focuses on configuring access log sinks and enabling Envoy to send access log messages via Contour.
+
+## Goals
+- Enable users to specify ALS configuration
+- Enable users to send HTTP access logs for specific listeners 
+
+## Non Goals
+- TCP Access Log Support
+
+## High-Level Design
+Contour will add support for Envoy's HTTP Access Log Service.
+ALS information will be specified as a static cluster for Envoy and Contour controller will set `access_log` field for each HTTPConnectionManager filter.
```

Comment:
It looks like Envoy requires the ALS cluster to be static, not a CDS cluster, is that right? If so, a brief explanation of that as a constraint (and some links) would be helpful since that guides the design.

Reply:
+1